daemon/cmd/daemon_main.go

```diff
@@ -1420,6 +1420,10 @@ func initEnv(vp *viper.Viper) {
 		}
 	}
 
+	if option.Config.EnableIPSec && option.Config.TunnelingEnabled() && option.Config.IPAM == ipamOption.IPAMENI {
+		log.Fatalf("IPAM ENI with IPSEC require routingMode=%s (--routing-mode=%s)", option.RoutingModeNative, option.RoutingModeNative)
+	}
```

Comment:
This isn't actually specific to IPsec. ENI IPAM mode doesn't make sense with tunneling routing mode, regardless of other features.

(You can read about those different modes in https://docs.cilium.io/en/stable/network/concepts/routing/#aws-eni-datapath. In particular, notice the pod IPs in ENI IPAM mode are directly routable on the AWS VPC and how the implementation via Linux rules & routes assumes traffic flows through the ENI devices and not some overlay.)

Reply:
As follow-up for https://github.com/cilium/cilium/pull/34651#pullrequestreview-2276056672, after a brief discussion it seems that there's no strong reason to support overlay with IPAM ENI, at least at the moment. Running such a config allows pod-to-pod, node-to-node and node-to-world communications, while not supporting pod-to-world ones. Enabling IPSec would cause also pod-to-pod to be disrupted, and before the 1st commit of this PR it would also lead to segfault.

Waiting to gather more feedbacks before proceeding, does it make sense at this point to check and block this config (2nd commit)?